amm/contracts/stable_pool/lib.rs

```diff
@@ -302,12 +302,12 @@ pub mod stable_pool {
             token_in: AccountId,
             token_out: AccountId,
         ) -> Result<(usize, usize), StablePoolError> {
+            if token_in == token_out {
```

Comment:
This still should work as intended however in case the `token_in` and `token_out` are identical addresses but they are not token addresses, it would return a misleading error message (they are identical true, but not token addresses).

Reply:
Sure, it still should work as intended